spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/MessageChannelConfigurerTests.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.cloud.stream.binder.redis.config.RedisMessageChannelBinderConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.junit.redis.RedisTestSupport;
+import org.springframework.cloud.stream.tuple.Tuple;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.Assert;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration({MessageChannelConfigurerTests.TestSource.class,
+		MessageChannelConfigurerTests.TestSink.class})
+public class MessageChannelConfigurerTests {
+
+	@Rule
+	public RedisTestSupport redisTestSupport = new RedisTestSupport();
+
+	@Autowired @Bindings(TestSink.class)
+	private Sink testSink;
+
+	@Autowired
+	@Bindings(TestSource.class)
+	private Source testSource;
+
+	@Test
+	public void testContentTypeConfigurer() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(1);
+		testSink.input().subscribe(new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				Assert.isTrue(message.getPayload() instanceof Tuple);
+				Assert.isTrue(((Tuple)message.getPayload()).getFieldNames().get(0).equals("message"));
+				Assert.isTrue(((Tuple) message.getPayload()).getValue(0).equals("Hi"));
+				latch.countDown();
+			}
+		});
+		testSource.output().send(MessageBuilder.withPayload("{\"message\":\"Hi\"}").build());
+		latch.await(10, TimeUnit.SECONDS);
+	}
+
+	@Test
+	public void testHistoryTrackerConfigurer() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(1);
+		testSink.input().subscribe(new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				Assert.isTrue(message.getHeaders().containsKey("SPRING_CLOUD_STREAM_HISTORY"),
+						"Message header should have tracking history info");
+				BindingProperties inputBindingProps = (BindingProperties) ((Map)((List) message.getHeaders()
+						.get("SPRING_CLOUD_STREAM_HISTORY")).get(0)).get("input");
+				Assert.isTrue(inputBindingProps.getDestination().equals("configure"));
+				Assert.isTrue(inputBindingProps.getTrackHistory());
+				BindingProperties outputBindingProps = (BindingProperties) ((Map)((List) message.getHeaders()
+						.get("SPRING_CLOUD_STREAM_HISTORY")).get(0)).get("output");;
+				Assert.isTrue(outputBindingProps.getDestination().equals("configure"));
+				Assert.isTrue(!outputBindingProps.getTrackHistory());
```

Comment:
Is there any particular reason you are using Spring's `Assert` rather than JUnit?